cypress/pageObject/Patients/PatientLocation.ts

```diff
@@ -0,0 +1,229 @@
+    cy.verifyAndClickElement(this.selectors.locationTab, "Location");
+    return this;
+  }
+
+  clickAddLocation() {
+    cy.verifyAndClickElement(this.selectors.addLocationButton, "Add Location");
+    return this;
+  }
+
+  selectLocationForm(locationForm: string) {
+    if (locationForm) {
+      cy.clickAndSelectOption(this.selectors.locationForm, locationForm);
+    }
+    return this;
+  }
+
+  enterLocationName(name: string, clearBeforeTyping: boolean = false) {
+    cy.typeIntoField(this.selectors.nameInput, name, { clearBeforeTyping });
+    return this;
+  }
+
+  enableBulkCreation(enableBulkCreation: boolean) {
+    if (enableBulkCreation) {
+      cy.get(this.selectors.enableBulkCreation).click();
+    }
+    return this;
+  }
+
+  selectBedCounts(bedCounts: string) {
+    if (bedCounts) {
+      cy.clickAndSelectOption(this.selectors.bedCountsSelect, bedCounts);
+    }
+    return this;
+  }
+
+  enterDescription(description: string) {
+    if (description) {
+      cy.typeIntoField(this.selectors.locationDescription, description);
+    }
+    return this;
+  }
+
+  selectLocationStatus(status: string) {
+    if (status) {
+      cy.clickAndSelectOption(this.selectors.locationStatus, status);
+    }
+    return this;
+  }
+
+  selectOperationalStatus(status: string) {
+    if (status) {
+      cy.clickAndSelectOption(this.selectors.operationalStatus, status);
+    }
+    return this;
+  }
+
+  fillLocationData(data: LocationData) {
+    this.selectLocationForm(data.form)
+      .enterLocationName(data.name)
+      .enableBulkCreation(data.form === "Bed")
+      .selectBedCounts(data.bedsCount)
+      .enterDescription(data.description)
+      .selectLocationStatus(data.status)
+      .selectOperationalStatus(data.opStatus);
+    return this;
+  }
+
+  interceptLocationCreationRequest() {
+    cy.intercept("POST", `**/api/v1/facility/**`).as("createLocation");
+    return this;
+  }
+
+  verifyLocationCreationAPICall() {
+    cy.wait("@createLocation").its("response.statusCode").should("eq", 200);
+    return this;
+  }
+
+  submitLocationForm() {
+    cy.clickSubmitButton("Create");
+    return this;
+  }
+
+  assertLocationCreationSuccess() {
+    cy.verifyNotification("Location Created");
+    return this;
+  }
+
+  assertMultipleBedsCreationSuccess(bedCount: string) {
+    cy.verifyNotification(`${bedCount} created successfully`);
+    return this;
+  }
+
+  navigateToEncounters() {
+    cy.get('[data-sidebar="content"]').contains("Encounters").click();
+    return this;
+  }
+
+  clickPlannedEncounterFilter() {
+    cy.intercept("GET", "**/api/v1/encounter/**").as("getEncounters");
+    cy.verifyAndClickElement('[data-cy="planned-filter"]', "Planned");
+    cy.wait("@getEncounters", { timeout: 10000 }).then((interception) => {
+      expect(interception.request.url).to.include("status=planned");
+      expect(interception.response.statusCode).to.eq(200);
+    });
+    return this;
+  }
+
+  openFirstEncounterDetails() {
```

Comment:
```suggestion
  openFirstEncounterDetails() {
```

use already existing function, don't create duplicate ones